Automattic/wp-calypso
# Domains Management i1: Initial domains table header with sorting

Related to https://github.com/Automattic/dotcom-forge/issues/3423

## Proposed Changes
This PR setup initial sorting for the domains table by:
* Adding `domains-table-header` component in the domains table package
* Add  `DomainsTableColumn` type that reflects the structure of the domain table and the header
* Include sorting function that's associated with each column/header

As it is the data is not currently sorted as sorting works on the `DomainData` which is lazy loaded as rows come into view. This PR doesn't handle that, I suspect we will need to eventually keep track of the `DomainData` data when it's loaded in the parent of the DomainsTableRow so that we can sort it and re-render the rows with the sorted data.

For this PR we simple add the `domains` header with the chevron indicating it's sortable.

## Testing Instructions



* browse `domains/manage?flags=domains/management`
* Ensure the header shows up with a single `domains` column
* Click the `Domains` header and the chevron should toggle to indicate the sortDirection. 

Next:


## Pre-merge Checklist
